smartapps/smartthings/ifttt.src/ifttt.groovy

```diff
@@ -131,19 +131,69 @@ def update() {
 	def type = params.deviceType
 	def data = request.JSON
 	def devices = settings[type]
+	def device = settings[type]?.find { it.id == params.id }
 	def command = data.command
-
+	
 	log.debug "[PROD] update, params: ${params}, request: ${data}, devices: ${devices*.id}"
-	if (command) {
-		def device = devices?.find { it.id == params.id }
-		if (!device) {
-			httpError(404, "Device not found")
-		} else {
-			device."$command"()
-		}
+	
+	if (!device) {
+		httpError(404, "Device not found")
```

Comment:
I am not sure if code stops executing after you call httpError(), you might want to confirm that.